test/spec/patient-search/controllers/patient-search.controller.test.js

```diff
@@ -0,0 +1,99 @@
+/*
+ jshint -W098, -W117, -W030
+ */
+(function() {
+  'use strict';
+  describe('Controller: PatientSearchCtrl ', function(){
+    var controller;
+    var scope;
+    var filterFilter;
+    var state;
+    var openmrsRestService;
+    var log;
+
+
+    beforeEach(function(){
+      module('ngAmrsApp');
+      module('app.openmrsRestServices');
+      module('app.patientsearch');
+      module('mock.data');
+
+    });
+
+    beforeEach(inject(function($controller, $injector, $rootScope, OpenmrsRestService){
+
+      //Injecting required dependencies
+      scope =   $rootScope.$new();
+      state = $injector.get('$state');
+      openmrsRestService = $injector.get('OpenmrsRestService');
+      log=$injector.get('$log');
+      filterFilter=$injector.get('filterFilter');
+
+      controller =$controller('PatientSearchCtrl', {
+        $rootScope:scope,
+        OpenmrsRestService:openmrsRestService,
+        $scope:scope,
+        $log:log,
+        filterFilter:filterFilter,
+        $state:state
+
+      });
+
+    }));
+
+      it('PatientSearchCtrl controller should be created successfully', function() {
+        expect(controller).to.exist;
+      });
+
+      it('should ensure main variables needed for the controller exists', function(){
+        expect(scope.patients).to.exist;
+        expect(scope.isBusy).to.exist;
+        expect(scope.noOfPages).to.exist;
+        expect(scope.currentPage).to.exist;
+        expect(scope.entryLimit).to.exist;
+
+      });
+      it('should ensure main variables needed for the controller exists', function(){
+
+        expect(scope.isBusy).to.be.equal(false);
+        expect(scope.currentPage).to.be.equal(1);
+        expect(scope.entryLimit).to.be.equal(10);
+
+      });
+      it('should ensure loadPatient method is defined', function(){
+          expect(scope.loadPatient).to.exist;
+          //expect(scope.loadPatient).toBeDefined();
+        });
+
+      it('should ensure resetFilters method is defined', function(){
+        expect(scope.resetFilters).to.exist;
+       // expect(scope.loadPatient).toBeDefined()
+      });
+
+      it('should return object when search method is fired', function(){
+        var searchString="test";
+        openmrsRestService.getPatientService().getPatientQuery({q:searchString},
+          function(data) {
+            var result=data;
+            expect(result).to.be.an('object');
+          });
+
+      });
+      it('should set isBusy to false when searchPatient is invoked and result is returned', function () {
+        scope.searchString="test";
+        scope.searchPatient();
+        expect(scope.isBusy).to.equal(false);
+
+      });
+      it('should return empty result when searchPatient is invoked with empty searchString', function () {
+        scope.searchString="";
+        scope.searchPatient();
+        expect(scope.patients.length).to.be.equal(0);
+      });
+
+    });
+
+
+
+
```

Comment:
Get rid of these blank lines and the one I mentioned above and we will be ready to merge